probe-rs/src/debug/language.rs

```diff
@@ -61,7 +54,19 @@ pub trait ProgrammingLanguage {
     }
 }
 
-#[derive(Clone)]
-pub struct UnknownLanguage;
+#[derive(Debug, Clone)]
+pub struct UnknownLanguage(DwLang);
```

Comment:
Storing the language means we have to allocate every time `from_dwarf` is called with something that's not Rust or C. In the grand scheme of things this isn't a huge issue, just something to be aware of.

[godbolt example](https://rust.godbolt.org/z/KsTK31nsx)

Reply:
Right now we're already returning a `Box` in line 16 in this file, so there isn't really a difference. I generally prefer improved error handling, and don't mind paying the cost of some allocations for that.